src/argilla_sdk/settings/_field.py

```diff
@@ -116,11 +116,28 @@ def title(self) -> Optional[str]:
     def name(self) -> str:
         return self._model.name
 
+    @property
+    def description(self) -> Optional[str]:
+        # TODO: Setting resources should be aligned at the API level
+        return None
+
+    @property
+    def required(self) -> bool:
+        # TODO: Setting resources should be aligned at the API level
+        return False
+
+    @property
+    def type(self) -> str:
+        # TODO: Setting resources should be aligned at the API level
+        return "vector"
+
 
 FieldType = Union[TextField, VectorField, MetadataField]
 
+_FieldModel = Union[TextFieldModel, VectorFieldModel, MetadataFieldModel]
+
 
-def field_from_model(model: FieldModel) -> FieldType:
+def field_from_model(model: _FieldModel) -> FieldType:
```

Comment:
Since this is only used below, we could lose the declaration of `_FieldModel`.

```suggestion


def field_from_model(model: Union[TextFieldModel, VectorFieldModel, MetadataFieldModel]) -> FieldType:
```